apps/platform/src/app/[chainId]/projects/propose/page.tsx

```diff
@@ -33,63 +29,91 @@ const getErrorMessage = (
 export default function CreateProjectPage() {
 	const t = useTranslations('proposalForm');
 
+	const {
+		selectedFiles,
+		handleFileChange,
+		openFilePicker,
+		removeFile,
+		inputRef,
+	} = useFileHandler();
+
 	const [isChecked, setIsChecked] = useState(false);
 	const [seconds, setSeconds] = useState(15);
+	const [isLoaded, setIsLoaded] = useState(false);
+	const [proposalFormData, setProposalFormData] = useState({
+		project: '',
+		location: '',
+		description: '',
+		name: '',
+		proposer: '',
+		email: '',
+		mobile: '',
+		isChecked: false,
+	});
 
 	const handleCheckboxChange = () => {
 		setIsChecked(prevState => !prevState);
+		setProposalFormData(prevData => ({
+			...prevData,
+			isChecked: !prevData.isChecked,
+		}));
 	};
 
 	const [state, formAction] = useFormState(createProjectAction, {
 		message: [],
 		status: false,
 	});
 
-	const [selectedFiles, setSelectedFiles] = useState<FileWithPreview[]>([]);
-	const inputRef = useRef<HTMLInputElement | null>(null);
-
-	const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-		const files: File[] = event.target.files
-			? Array.from(event.target.files)
-			: [];
-
-		// Map through files and create a preview URL for each file
-		const filePreviews: FileWithPreview[] = files.map(file => ({
-			file,
-			url: URL.createObjectURL(file),
-		}));
-
-		setSelectedFiles(prevFiles => [...prevFiles, ...filePreviews]);
-	};
-
-	const handleButtonClick = () => {
-		if (inputRef.current) {
-			inputRef.current.click();
+	// Load data from localStorage when the component mounts
+	useEffect(() => {
+		try {
+			const storedData = JSON.parse(
+				localStorage.getItem('proposalFormData') || '{}',
+			);
+			setProposalFormData({
```

Comment:
How about 
```javascript
if (storedData) setProposalFormData(storedData);
```
The `useState` will make sure that all the props are present and set to `''` on init. No need to 'destructure' and check for defaults.

Reply:
thx, junior mistake :)

Reply:
not junior mistake :D, now I remember it is about typescript issue, typescript looks at it like undefined on input default values I refactor code on other way now it's ok ;)